pilkit/processors/utils.py

```diff
@@ -19,4 +19,15 @@ def histogram_entropy_py(image):
 
 # Select the Pillow native histogram entropy function - if
 # available - and fall back to the Python implementation:
-histogram_entropy = getattr(Image.Image, 'entropy', histogram_entropy_py)
\ No newline at end of file
+histogram_entropy = getattr(Image.Image, 'entropy', histogram_entropy_py)
+
+def resolve_palette(image):
+    """ Convert a palette image to a non-palette image. """
+
+    # Work around a bug in PIL: Image.palette doesn't check to see if
```

Comment:
I'm not sure if this is a bug or intended behaviour. If you read the source of the method [`getpalette()`](https://pillow.readthedocs.io/en/stable/_modules/PIL/Image.html#Image.getpalette) you will see that it also load the image before accessing some attributes.

We can say simply that "We need to load the image before accessing the palette" and call it a day ;)

Reply:
That makes sense. Does it look ok for you now?